packages/rpc-provider/src/ws/index.ts

```diff
@@ -108,7 +108,7 @@ export class WsProvider implements ProviderInterface {
    * @param {Record<string, string>} headers The headers provided to the underlying WebSocket
    * @param {number} [timeout] Custom timeout value used per request . Defaults to `DEFAULT_TIMEOUT_MS`
    */
-  constructor (endpoint: string | string[] = defaults.WS_URL, autoConnectMs: number | false = RETRY_DELAY, headers: Record<string, string> = {}, timeout?: number) {
+  constructor (endpoint: string | string[] = defaults.WS_URL, autoConnectMs: number | false = RETRY_DELAY, headers: Record<string, string> = {}, cacheCapacity: number = DEFAULT_CAPACITY, timeout?: number) {
```

Comment:
Rather move this as the last parameter. (No breaking change - we really don't like this at all, unless very much required)

On use can do `this.#callCache = new LRUCache(cacheCapacity || DEFAULT_CAPACITY);` to apply defaults.

Reply:
Thanks for the feedback. I just applied the change you recommended and I agree it makes more sense.